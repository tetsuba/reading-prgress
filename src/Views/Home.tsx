import { useState } from 'react'
import { Link, useNavigate } from 'react-router-dom'
import Modal from '../Components/Modal/Modal'
import Login from './Login'
import H1 from '../Components/H1/H1'
import P from '../Components/P/P'
import Button from '../Components/Button/Button'
import Svg from '../Components/Svg/Svg'
import Display from '../Components/Dispay/Display'

const Home = () => {
    const navigate = useNavigate()
    const [showLogin, setShowLogin] = useState(false)
    return (
        <main>
            <div className="px-6 lg:px-8">
                <div className="mx-auto max-w-3xl py-10 sm:py-40">
                    <div className="mb-8 hidden justify-center sm:flex ">
                        <div className="rounded-full px-4 py-1.5 text-sm ring-1 ring-gray-900/10 hover:ring-gray-900/20">
                            <span className="text-gray-600">
                                Decodable Reading Books For Children
                            </span>
                            <Link
                                to=""
                                className="pl-1 font-semibold text-indigo-600"
                            >
                                Read more{' '}
                                <Svg
                                    icon="arrowRight"
                                    className="inline-block"
                                />
                            </Link>
                        </div>
                    </div>
                    <div>
                        <H1>Data to improve reading</H1>
                        <P className="mt-6 text-lg leading-8 text-gray-600 sm:text-center">
                            Reading Tracker helps to identify the weaknesses of
                            a child learning to read. It will help identify
                            which phonetics the reader is struggling with and
                            which sight words it cannot read.
                        </P>
                        <div className="mt-8 flex gap-x-4 sm:justify-center">
                            <Button
                                className="flex items-center"
                                onClick={() => setShowLogin(true)}
                                template="primary"
                            >
                                Log in{' '}
                                <Svg icon="arrowRight" className="ml-1" />
                            </Button>
                            <Button
                                className="flex items-center"
                                onClick={() => navigate('/register')}
                                template="secondary"
                            >
                                Register{' '}
                                <Svg icon="arrowRight" className="ml-1" />
                            </Button>
                        </div>
                    </div>
                </div>
            </div>
            <Display value={showLogin}>
                <Modal className="md:max-w-xl">
                    <Login setShowLogin={setShowLogin} />
                </Modal>
            </Display>
        </main>
    )
}

export default Home
