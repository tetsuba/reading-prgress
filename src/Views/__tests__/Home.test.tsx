import {
    render,
    fireEvent,
    screen,
    waitFor,
    waitForElementToBeRemoved
} from '@testing-library/react'
import Home from '../Home'
import {
    WrapperWithRouter,
    WrapperWith_Store_Query_Router
} from '../../vitest-setup'

const mockNavigate = vi.fn()
vi.mock('react-router-dom', async () => {
    const actual = await vi.importActual('react-router-dom')
    return {
        ...(actual as object),
        useNavigate: () => mockNavigate
    }
})

describe('Home', () => {
    describe('@Render', () => {
        test('should render first load', () => {
            const { asFragment } = render(
                <WrapperWithRouter pathname="/">
                    <Home />
                </WrapperWithRouter>
            )
            expect(asFragment()).toMatchSnapshot()
        })

        test('should open and close login modal', async () => {
            render(
                <WrapperWith_Store_Query_Router pathname="/">
                    <Home />
                </WrapperWith_Store_Query_Router>
            )
            fireEvent.click(screen.getByText('Log in'))
            await waitFor(() => expect(screen.getByTestId('login-view')))
            fireEvent.click(screen.getByTestId('modal-close'))
            expect(screen.queryByTestId('login-view')).toBeNull()
        })

        test('clicking on the registration button', async () => {
            render(
                <WrapperWithRouter pathname="/">
                    <Home />
                </WrapperWithRouter>
            )
            fireEvent.click(screen.getByText(/^Register/))
            expect(mockNavigate).toHaveBeenCalled()
        })
    })
})
