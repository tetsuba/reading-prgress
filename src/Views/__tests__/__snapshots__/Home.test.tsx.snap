// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > @Render > should render first load 1`] = `
<DocumentFragment>
  <main>
    <div
      class="px-6 lg:px-8"
    >
      <div
        class="mx-auto max-w-3xl py-10 sm:py-40"
      >
        <div
          class="mb-8 hidden justify-center sm:flex "
        >
          <div
            class="rounded-full px-4 py-1.5 text-sm ring-1 ring-gray-900/10 hover:ring-gray-900/20"
          >
            <span
              class="text-gray-600"
            >
              Decodable Reading Books For Children
            </span>
            <a
              class="pl-1 font-semibold text-indigo-600"
              href="/"
            >
              Read more 
              <svg
                class="h-4 w-4 inline-block"
                fill="none"
                stroke="currentColor"
                stroke-width="2.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.5 12h15m0 0l-6.75-6.75M19.5 12l-6.75 6.75"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </a>
          </div>
        </div>
        <div>
          <h1
            class="text-4xl font-bold tracking-tight sm:text-center sm:text-6xl "
          >
            Data to improve reading
          </h1>
          <p
            class="mt-6 text-lg leading-8 text-gray-600 sm:text-center"
          >
            Reading Tracker helps to identify the weaknesses of a child learning to read. It will help identify which phonetics the reader is struggling with and which sight words it cannot read.
          </p>
          <div
            class="mt-8 flex gap-x-4 sm:justify-center"
          >
            <button
              class="rounded-lg font-medium text-base text-center px-5 py-2.5 text-white bg-blue-700 hover:bg-blue-800 flex items-center"
              template="primary"
            >
              Log in 
              <svg
                class="h-4 w-4 ml-1"
                fill="none"
                stroke="currentColor"
                stroke-width="2.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.5 12h15m0 0l-6.75-6.75M19.5 12l-6.75 6.75"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
            <button
              class="rounded-lg font-medium text-base text-center px-5 py-2.5 text-gray bg-gray-200 hover:bg-gray-300 flex items-center"
              template="secondary"
            >
              Register 
              <svg
                class="h-4 w-4 ml-1"
                fill="none"
                stroke="currentColor"
                stroke-width="2.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.5 12h15m0 0l-6.75-6.75M19.5 12l-6.75 6.75"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
